### DISCUSSION TITLE
Dim background on dialog opened from drawer

### DISCUSSION THREAD

**Discussion by 15MariamS** [microsoft/fluentui#31305] [Q&A] [answered]
When I open a dialog from an open drawer, the background is not dimmed. Presumably, this is because the drawer is also a type of dialog. How can I ensure that the background still dims?

What I expect when I open a dialog:
<img width="451" alt="Screenshot 2024-05-07 at 3 13 56 PM" src="https://github.com/microsoft/fluentui/assets/81782111/46065195-e794-4cc3-9de0-63d5edfecc2f">

**Comment by ValentinaKozlova** [comment]
@15MariamS , could you please provide more details on how you use Drawer? A link on codesandbox would be helpful.

**Reply by 15MariamS** [reply]
[Here is a link](https://stackblitz.com/edit/mw9wzn-cm2okk?file=src%2Fexample.tsx). The background dims if the dialog is called from _outside_ the overlay, but if it's called from within the overlay, then the background doesn't dim.

**Actual (left) vs expected (right)**
<img width="328" alt="Screenshot 2024-05-20 at 2 21 24 PM" src="https://github.com/microsoft/fluentui/assets/81782111/f2a7d17f-e8bf-4bc1-ab06-e1f7c7655e03">   <img width="305" alt="Screenshot 2024-05-20 at 2 21 47 PM" src="https://github.com/microsoft/fluentui/assets/81782111/a39460bd-6c91-4244-bbac-501cf19ff463">

**Reply by khmakoto** [reply, marked as answer]
@15MariamS this happens indeed because we set the background of the `backdrop` to transparent if you have nested dialogs. Is there a reason why you need to call the `Dialog` inside of the `OverlayDrawer`? I'm asking because you basically have a work-around solution in the link you shared.

If there is a reason why you absolutely have to open the `Dialog` from within the `OverlayDrawer`, then you can customize the styles of the backdrop to the color used when you don't have nested dialogs, that is `backgroundColor: 'rgba(0, 0, 0, 0.4)'`. You can take a look at the solution in [this modified stackblitz](https://stackblitz.com/edit/mw9wzn-lqnimc?file=src%2Fexample.tsx) based on the link you shared.

**Reply by 15MariamS** [reply]
Thank you so much for the help, @khmakoto!

**Reply by khmakoto** [reply]
No prob! Glad that the solution worked for you!

**Reply by 15MariamS** [reply]
@khmakoto in the modified stackblitz that you linked above, the backdrop class is memoized. What happens if we don't memoize? Doesn't look like the calculation is time consuming.

Note: I do notice I get the classname incorrectly showing up in the background when I _don't_ use useMemo... just curious why it's necessary.
<img width="373" alt="Screenshot 2024-09-23 at 4 20 14 PM" src="https://github.com/user-attachments/assets/4d217871-1eea-4f17-92a2-f87ad8e2b80c">

**Reply by khmakoto** [reply]
It's mostly assuming you have other things that you're passing as part of the backdrop props, but I agree that it's not time consuming if it's only the styles.

I don't get the classname showing up in the background when I remove `useMemo` by the way. Do you have an example where that happens?
